config.go

```diff
@@ -289,7 +289,8 @@ type Config struct {
 	ExternalIPs       []net.Addr
 	DisableListen     bool          `long:"nolisten" description:"Disable listening for incoming peer connections"`
 	DisableRest       bool          `long:"norest" description:"Disable REST API"`
-	DisableRestTLS    bool          `long:"no-rest-tls" description:"Disable TLS for REST connections"`
+	DisableTLS        bool          `long:"notls" description:"Disable TLS encryption for RPC"`
```

Comment:
Wouldn't it make more sense to call the new option `no-grpc-tls` and change it so the two flags don't overlap? So if one wants to disable both, that would mean indicating both flags.

Reply:
Hm possibly although to me it seems a bit odd to disable it on one but not the other? From what I can tell the reason the `DIsableRestTLS` option was added was to avoid having to make changes to lncli, not because there's a practical reason to have TLS on RPC but not REST (or vice-versa). I figure the users of the `DisableRestTLS` option are using REST almost exclusively. I just don't see a point in doing part TLS, part non-encrypted, since if there's a successul MITM attack on only the non-encrypted part the attacker still gets your macaroon.

Reply:
...And the authors and reviewers of the PR of the existing flag did not see a point in disabling it for gRPC at all, yet here we are ;)
All other things being equal, a consistent API and path-of-least-surprise should be preferable IMO - less cognitive overhead for devs and users alike.
That aside, perhaps there will be users with scenarios where that currently seemingly unlikely need does arise, which would necessitate a *third* flag...

Reply:
> And the authors and reviewers of the PR of the existing flag did not see a point in disabling it for gRPC at all, yet here we are

That PR did disable TLS for both gRPC & REST originally, fyi, and was changed after the issue with `lncli` was brought up in reviews.

IMO the simplest/most intuitive thing is to disable TLS altogether but that's just my 2 sats.

Reply:
Because disabling TLS for gRPC has the technical consequences discussed below, I think it makes sense to have the two flags separate. And the gRPC one should have a comment in its description with something along the lines of "Requires the client to be h2c compatible. See docs/grpc-h2c.md".